### PR TITLE
fix: install application and list application metadata

### PR DESCRIPTION
**Pull request by frdomovic**
# fix: install application and list application metadata

## Summary:
As the logic for application installation has changed in the Admin API, the function arguments for installation and response type has changed.
In this PR the api response type is updated and also the UI components data fetching was updated.

### CODE REVIEW

File: src/api/dataSource/NodeDataSource.ts [hunk 1]
```diff
@@ -31,6 +31,14 @@ export interface Application {
   version: string;
 }
 
+export interface BlobApplication {
```
Comment (MatejVukosav) [inline src/api/dataSource/NodeDataSource.ts:34]: What is difference between Application and BlobApplication?

File: src/pages/StartContext.tsx [hunk 2]
```diff
@@ -69,9 +68,9 @@ export default function StartContextPage() {
     setShowStatusModal(true);
   };
 
-  const installApplicationHandler = async (): Promise<boolean> => {
+  const installApplicationHandler = async (): Promise<string> => {
```
Comment (MatejVukosav) [inline src/pages/StartContext.tsx:71]: `string | null`

File: src/pages/StartContext.tsx [hunk 3]
```diff
@@ -69,9 +68,9 @@ export default function StartContextPage() {
...
     if (!application.appId || !application.version) {
-      return false;
+      return '';
```
Comment (MatejVukosav) [inline src/pages/StartContext.tsx:73]: What do we do with '' ?

File: src/api/dataSource/NodeDataSource.ts (hunk 1 above)
Comment (frdomovic) [reply src/api/dataSource/NodeDataSource.ts:34]: BlobApplication is the updated object from admin api and I still want to leave Application available as we might need to update it again

File: src/pages/StartContext.tsx (hunk 3 above)
Comment (frdomovic) [reply src/pages/StartContext.tsx:73]: can do null, in other function it checks if appId is "" then returns

File: src/api/dataSource/NodeDataSource.ts (hunk 1 above)
Comment (frdomovic) [reply src/api/dataSource/NodeDataSource.ts:34]: Screenshot 2024-08-08 at 11 59 11

File: src/pages/StartContext.tsx (hunk 2 above)
Comment (frdomovic) [reply src/pages/StartContext.tsx]: Updated

File: src/pages/StartContext.tsx (hunk 3 above)
Comment (frdomovic) [reply src/pages/StartContext.tsx]: Updated

File: src/api/dataSource/NodeDataSource.ts (hunk 1 above)
Comment (MatejVukosav) [reply src/api/dataSource/NodeDataSource.ts:34]: But can we use Application for both?

Comment (frdomovic) [reply src/api/dataSource/NodeDataSource.ts:34]:
```
BlobApplication {
  id: string;
  blob: string;
  version: string;
  source: string;
  contract_app_id: string;
}
```

```
interface Package {
  id: string;
  name: string;
  description: string;
  repository: string;
  owner: string;
}

interface Application extends Package {
  version: string;
  contract_app_id: string;
}
```
I would leave it as is

File: src/api/dataSource/NodeDataSource.ts [hunk 4]
```diff
@@ -298,24 +310,25 @@ export class NodeDataSource implements NodeApi {
...
           hash,
-        },
-        headers ?? {},
+        }),
       );
+      const response: ResponseData<InstallApplicationResponse> =
+        await this.client.post<InstallApplicationResponse>(
+          `${getAppEndpointKey()}/admin-api/install-application`,
+          {
+            contract_app_id: selectedPackage,
```
Comment (MatejVukosav) [inline src/api/dataSource/NodeDataSource.ts:326]: `selectedPackage` does not indicate a connection with `contract_app_id`. 

Can you expand the update to the naming?

Comment (frdomovic) [reply src/api/dataSource/NodeDataSource.ts]: selectedPackageId updated

File: src/api/dataSource/NodeDataSource.ts (hunk 1 above)
Comment (frdomovic) [reply src/api/dataSource/NodeDataSource.ts]: Updated

File: src/api/dataSource/NodeDataSource.ts [hunk 5]
```diff
@@ -28,7 +28,14 @@ export interface User {
 
 export interface Application {
   id: string;
+  blob: string;
   version: string;
+  source: string;
+  contract_app_id: string;
+  name: string | undefined;
```
Comment (MatejVukosav) [inline src/api/dataSource/NodeDataSource.ts:35]: string | null

File: src/api/dataSource/NodeDataSource.ts [hunk 6]
```diff
@@ -95,7 +102,7 @@ export interface DidResponse {
   did: Did;
 }
 
-export interface ListApplicationsResponse {
+export interface ListApplicationResponse {
```
Comment (MatejVukosav) [inline src/api/dataSource/NodeDataSource.ts:105]: GetInstalledApplicationsResponse

File: src/pages/ApplicationDetails.tsx [hunk 7]
```diff
@@ -26,7 +26,7 @@ export default function ApplicationDetailsPage() {
         const packageData = await getPackage(id);
         const versionData = await getReleases(id);
         setApplicationInformation({
-          package: packageData,
+          package: packageData as Package,
```
Comment (MatejVukosav) [inline src/pages/ApplicationDetails.tsx:29]: Assure object has correct type before using

File: src/pages/StartContext.tsx [hunk 8]
```diff
@@ -40,18 +40,15 @@ export default function StartContextPage() {
 
   const startContext = async () => {
     setIsLoading(true);
-    const response = await installApplicationHandler();
-    if (!response) {
+    const appId = await installApplicationHandler();
```
Comment (MatejVukosav) [inline src/pages/StartContext.tsx:43]: Add type

File: src/api/dataSource/NodeDataSource.ts (hunk 5 above)
Comment (frdomovic) [reply src/api/dataSource/NodeDataSource.ts]: updated

File: src/api/dataSource/NodeDataSource.ts (hunk 6 above)
Comment (frdomovic) [reply src/api/dataSource/NodeDataSource.ts]: updated

File: src/pages/ApplicationDetails.tsx (hunk 7 above)
Comment (frdomovic) [reply src/pages/ApplicationDetails.tsx]: Added if check before set state as packageData is always Package or null

File: src/pages/StartContext.tsx (hunk 8 above)
Comment (frdomovic) [reply src/pages/StartContext.tsx]: added

**Review by MatejVukosav — APPROVED** [review]

**Pull request merged**